CMakeLists.txt

```diff
@@ -78,10 +77,26 @@ set(cryptopp_SOURCES
 		${cryptopp_SOURCES}
 		)
 
-if(MINGW)
+if(MINGW OR WIN32)
 	list(APPEND cryptopp_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/winpipes.cpp)
 endif()
 
+if(MSVC AND NOT DISABLE_ASM)
+ 	if(CMAKE_CL_64) 		
+ 		list(APPEND cryptopp_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/x64dll.asm)
+ 		list(APPEND cryptopp_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/x64masm.asm)
+        set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/x64dll.asm PROPERTIES COMPILE_FLAGS "/D_M_X64")
+        set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/x64masm.asm PROPERTIES COMPILE_FLAGS "/D_M_X64")
+        set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/rdrand.asm PROPERTIES COMPILE_FLAGS "/D_M_X64")
+ 	else()    
+        set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/x64dll.asm PROPERTIES COMPILE_FLAGS "/D_M_X86")
+        set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/x64masm.asm PROPERTIES COMPILE_FLAGS "/D_M_X86")
+        set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/rdrand.asm PROPERTIES COMPILE_FLAGS "/D_M_X86")

```

Comment:
The pickup of `/D_M_X86` and `/D_M_X64` was good. However, there's a subtle issue remaining: both `rdrand.cpp` and `rdrand.asm` produce an outfile of of `rdrand.obj`. Here's how we handle it in the [test nmake makefile](http://github.com/weidai11/cryptopp/blob/master/cryptest.nmake#L119):

* `rdrand.cpp` produces output artifact `rdrand.obj`
* `rdrand.asm` with `/D_M_X86` produce output artifact `rdrand-x86.obj`
* `rdrand.asm` with `/D_M_X64` produce output artifact `rdrand-x64.obj`

Then, link with the pair (`rdrand.obj`, `rdrand-x86.obj`) or (`rdrand.obj`, `rdrand-x64.obj`)